app/models/vacancy.rb

```diff
@@ -28,4 +30,22 @@ class Vacancy < ApplicationRecord
     shared_room: 1,
     entire_residence: 2
   }
+
+  aasm column: :availability, enum: true do
+    state :available
+    state :available_soon
+    state :unavailable
+
+    event :vacate_soon do
+      transitions from: %i[unavailable, available], to: :available_soon
```

Comment:
```suggestion
      transitions from: %i[unavailable available], to: :available_soon
```
I might be wrong, but this comma isn't necessary here, is it?